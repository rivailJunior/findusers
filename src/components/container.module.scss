.mainContainer {
    // position: absolute;
    display: grid;
    height: auto;

    // background-color: #eaeaead6;
    grid-template-columns: repeat(12, 1fr);
    align-self: center;
    margin-right: 15%;
    margin-left: 15%;
}

@mixin grid-column($columnSize) {
    grid-column: span $columnSize;
    min-height: 50px;
    padding: 10px;
}

$list: 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12;

@each $size in $list {
    .containerItem#{$size} {
        @include grid-column($size);
    }
}

/* Smartphones (portrait and landscape) ----------- */
@media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    .mainContainer {
        margin-right: 1%;
        margin-left: 1%;
    }
    @each $size in $list {
        .containerItem#{$size} {
            @include grid-column(12);
        }
    }
}

/* iPads (portrait and landscape) ----------- */
@media only screen and (min-device-width: 481px) and (max-device-width: 1024px) {
}
