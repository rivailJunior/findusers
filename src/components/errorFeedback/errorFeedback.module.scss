@use '../../pages/index.scss' as index;

@mixin errorLabel($fontSize: 16) {
    @include index.default-label(14px, 1%, 100%, 500);
    font-size: $fontSize;
}

.errorContainer > div {
    @include errorLabel(20px);
    display: flex;
    flex-direction: column;
    width: 100%;
    min-height: 50px;
    color: white;
    background-color: lightcoral;
    justify-content: center;
    border-radius: 2px;
}

.subTitle {
    @include errorLabel(14);
}
