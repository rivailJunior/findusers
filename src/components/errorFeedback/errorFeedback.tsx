import React from 'react';
import styles from './errorFeedback.module.scss';

type iErrorFeedback = {
    title: string
    subTitle?: string
}
const ErrorFeedback = ({ title, subTitle }: iErrorFeedback) => {
    return (
        <div className={styles.errorContainer}>
            <div>{title}</div>
            {subTitle && <div data-testid="subTitleError" className={styles.subTitle}>{subTitle}</div>}
        </div>
    );
}

export { ErrorFeedback };
