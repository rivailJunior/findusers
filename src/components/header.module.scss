.header {
    overflow: hidden;
    background-color: white;
    padding: 10px 10px;
}

.header a {
    float: left;
    color: black;
    text-align: center;
    padding: 12px;
    text-decoration: none;
    font-size: 18px;
    line-height: 25px;
    border-radius: 4px;
}

.header a.logo {
    font-size: 25px;
    font-weight: bold;
}

.header a:hover {
    background-color: #ddd;
    color: black;
    cursor: pointer;
}

.header a.active {
    background-color: dodgerblue;
    color: white;
}

.header-right {
    float: right;
}

@media screen and (max-width: 500px) {
    .header a {
        float: none;
        display: block;
        text-align: left;
    }

    .header-right {
        float: none;
    }
}
