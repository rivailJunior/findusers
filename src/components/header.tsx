import React from 'react';
import styles from './header.module.scss'

type iHeader = {
    userName?: string,
    labelRight?: string,
    handleRight?(): Function
}
const Header = ({ userName, labelRight, handleRight }: iHeader): JSX.Element => {
    return (
        <div className={styles.header}>
            <a href="#default" className={styles.logo}>{userName}</a>
            {labelRight && (
                <div className={styles['header-right']}>
                    <a className={styles.active} onClick={handleRight} >Sair</a>
                </div>
            )}

        </div>

    );
}

export { Header };
