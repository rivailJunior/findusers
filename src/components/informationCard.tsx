import React from 'react';
import styles from './informationCard.module.scss';

type iCard = {
    title: string,
    mainText: string,

}
const InformationCard = ({ title, mainText }: iCard) => {
    return (
        <div className={styles.card}>
            <p>{title}</p>
            <h2>{mainText}</h2>
        </div>
    );
}

export { InformationCard };
