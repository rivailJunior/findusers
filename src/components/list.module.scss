@use '../pages/index.scss' as index;
@use '../pages/root.scss' as root;
.item {
    background-color: white;
    border: 0.5px solid #9e9e9e80;
    display: flex;
    flex-direction: column;
    overflow: hidden;
    border-radius: 5px;

    .itemLi {
        @include index.default-label(15px, 0, 100%, 300, 15px);
        display: grid;
        text-align: left;
        padding: 1%;
        display: grid;
        grid-template-columns: repeat(12, 1fr);
        max-height: 100px;

        .left {
            grid-column: span 2;
            display: flex;
            justify-content: center;
            align-self: center;
            font-size: 3rem;
        }
        .right {
            grid-column: span 10;

            .rightTitle {
                @include index.default-label(15px, 0, 100%, 700, 20px);
                // font-size: 1rem;
                text-transform: capitalize;
                text-align: left;
            }
            .rightSub {
                @include index.default-label(15px, 1%, 100%, 300, 20px, left);
                display: flex;
                flex-direction: column;
                justify-content: space-between;
            }
        }
    }
    .itemLi:hover {
        cursor: pointer;
        background-color: root.$hoverBgc;
        color: root.$hoverColor;
    }
}

/* Smartphones (portrait and landscape) ----------- */
@media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    .item {
        .itemLi {
            @include index.default-label(12px, 0, 100%, 500);

            grid-template-columns: repeat(12, 1fr);

            .left {
                display: none;
            }
            .right {
                grid-column: span 12;

                .rightTitle {
                    font-size: 1rem;
                    text-transform: capitalize;
                }
                .rightSub {
                    @include index.default-label(
                        10px,
                        1%,
                        100%,
                        300,
                        10px,
                        left
                    );
                    display: flex;
                    flex-direction: column;
                    justify-content: space-between;
                }
            }
        }
        .itemLi:hover {
            cursor: pointer;
            background-color: root.$hoverBgc;
            color: root.$hoverColor;
        }
    }
}

/* iPads (portrait and landscape) ----------- */
@media only screen and (min-device-width: 481px) and (max-device-width: 1024px) {
}

/* Desktops and laptops ----------- */
@media only screen and (min-width: 1224px) {
    /* Styles */
}
