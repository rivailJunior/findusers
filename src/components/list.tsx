import React from 'react';
import { ErrorFeedback } from './errorFeedback';
import styles from './list.module.scss';
import moment from 'moment'

type iList = {
    data: object[],
    handleClick: Function
}

const ListRepositories = ({ data, handleClick }: iList): JSX.Element => {
    return data.length ? (
        <div data-testid="list" className={styles.item}>
            {data.map((item: any, index) => {
                return (
                    <div key={index} className={styles.itemLi} onClick={() => handleClick(item)} data-testid="listItem">
                        <div className={styles.left}>
                            {item.stargazers_count}
                        </div>
                        <div className={styles.right}>
                            <div className={styles.rightTitle}>
                                {item?.name}
                            </div>
                            <div className={styles.rightSub}>
                                <div>{moment(item.created_at).format('DD/MM/YYYY')}</div>
                                <div> <a href={item.clone_url} target="_blank" rel="noopener noreferrer">{item.clone_url}</a></div>
                            </div>
                        </div>
                    </div>)
            })}
        </div>
    ) : (
        <ErrorFeedback title="Nada encontrado" />
    );
}

export { ListRepositories };
