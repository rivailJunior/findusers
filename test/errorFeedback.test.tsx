import React from 'react'
import { render, cleanup, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { ErrorFeedback } from '../src/components/errorFeedback'

afterEach(cleanup)

describe('Error Component', () => {
    test('Render Correctly', () => {
        const container = renderer.create(<
            ErrorFeedback title="Nenhum repositÃ³rio encontrado!" />)
        container.toJSON();
        expect(container).toMatchSnapshot()
    })

    test('Show with title and subtitle', () => {
        const { getByText } = render(<ErrorFeedback title="Nenhum usuario encotrado!" subTitle="Busque outro usuario" />)
        expect(getByText(/Nenhum usuario encotrado!/i)).toBeInTheDocument();
        expect(getByText(/Busque outro usuario/i)).toBeInTheDocument();
    })

    test('Show only title error message', () => {
        const { getByText, getByTestId } = render(<ErrorFeedback title="Nenhum usuario encotrado!" />)
        expect(getByText(/Nenhum usuario encotrado!/i)).toBeInTheDocument();
        expect(screen.queryByTestId('subTitleError')).not.toBeInTheDocument();
    })
})