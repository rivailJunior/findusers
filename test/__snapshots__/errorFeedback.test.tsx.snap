// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component Render Correctly 1`] = `
<div
  className="errorContainer"
>
  <div>
    Nenhum reposit√≥rio encontrado!
  </div>
</div>
`;
